src/core/tc_executor.cc

```diff
@@ -25,31 +28,23 @@ using namespace dlutils;
 const size_t TcExecutor::InvalidHandle;
 
 namespace {
-lang::TreeRef parseOneFunction(const std::string& def) {
-  lang::Parser parser(def);
-  auto r = parser.parseFunction();
-  if (parser.L.cur().kind != lang::TK_EOF) {
-    throw lang::ErrorReport(parser.L.cur().range)
-        << "More than one TCs were passed to TcExecutor.";
-  }
-  return r;
-}
-
 int toTypeToken(DLDataType dtype) {
   return lang::TypeInfo(lang::TypeInfo::Code(dtype.code), dtype.bits)
       .toScalarToken();
 }
 } // namespace
 
 TcExecutor::TcExecutor(
-    const std::string& TcDefinition,
-    const std::vector<const DLTensor*>& inputsInfo)
-    : TcExecutor(parseOneFunction(TcDefinition), inputsInfo) {}
```

Comment:
The removal of this constructor, along with `parseOneFunction`,
looks like an independent change.

Reply:
why is this constructor getting dropped? we want to allow taking `treeRef` or the string as the input. :)

Reply:
It's essentially dropped because it is not needed; a.k.a dead code.
We can resuscitate in the future if needed.

Reply:
@skimo-openhub this whole commit is about consolidating 2 classes and 3 constructors into 1 class and 1 constructor. I'd claim this is (and removing the dead code that results) is very much dependent change.
Makes sense?